src/Schemes/AffineSchemes/Objects/Attributes.jl

```diff
@@ -202,72 +396,32 @@ julia> (x1,x2,x3) = gens(R)
 julia> Y = subscheme(X,ideal(R,[x1*x2]))
 Spec of Quotient of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field by ideal(x1*x2)
 
-julia> defining_ideal(Y)
+julia> ambient_closure_ideal(Y)
 ideal(x1*x2)
 ```
 """
-@attr defining_ideal(X::AbsSpec{<:Any, <:MPolyRing}) = ideal(OO(X), [zero(OO(X))])
-defining_ideal(X::AbsSpec{<:Any, <:MPolyQuo}) = modulus(OO(X))
-@attr defining_ideal(X::AbsSpec{<:Any, <:MPolyLocalizedRing}) = ideal(OO(X), [zero(OO(X))])
-defining_ideal(X::AbsSpec{<:Any, <:MPolyQuoLocalizedRing}) = modulus(OO(X))
-
-
-@Markdown.doc """
-    strict_modulus(X::AbsSpec)
-
-This method return the strict modulus of an affine scheme ``X``.
-This is the ideal ``I`` in the `ambient_ring` of ``X`` consisting 
-of elements ``f ∈ R`` such that their restriction to ``X`` vanishes.
-
-# Examples
-```jldoctest
-julia> X = affine_space(QQ,3)
-Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
-
-julia> R = OO(X)
-Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
-
-julia> (x1,x2,x3) = gens(R)
-3-element Vector{fmpq_mpoly}:
- x1
- x2
- x3
-
-julia> Y = subscheme(X,ideal(R,[x1*x2]))
-Spec of Quotient of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field by ideal(x1*x2)
-
-julia> strict_modulus(Y)
-ideal(x1*x2)
-```
-"""
-strict_modulus(X::AbsSpec) = saturated_ideal(modulus(OO(X)))
-
-
-
-########################################################################
-# (3) Further attributes for Spec
-########################################################################
-
-strict_modulus(X::Spec) = saturated_ideal(modulus(OO(X)))
-
+@attr ambient_closure_ideal(X::AbsSpec{<:Any, <:MPolyRing}) = ideal(OO(X), [zero(OO(X))])
+ambient_closure_ideal(X::AbsSpec{<:Any, <:MPolyQuo}) = modulus(OO(X))
+@attr ambient_closure_ideal(X::AbsSpec{<:Any, <:MPolyLocalizedRing}) = ideal(OO(X), [zero(OO(X))])
```

Comment:
If it is the closure in the ambient affine space, then this lives in OO(ambient_affine_space) by naming convention and not in OO(X) in the localized setting. If this is to live in the localized ring (a setting for which such a command is needed as well !!), then we need a different name respecting the localization (which defaults to ambient_closure_ideal for non-localized rings).

Reply:
You are right. Here the previous implementation differs from the needs you formulated yesterday. I think this is up to you how to resolve this. Unless it breaks the code, of course.

Reply:
From my perspective, the only consistent way to do this is:
- ambient_... always refers to objects related to ambient_affine_space
- ambient_closure_ideal then has to provide an ideal in OO(ambient_affine_space)
- is there any good name for the localized equivalent or do we expect that users to use modulus(OO(X)) for that?

Reply:
> is there any good name for the localized equivalent or do we expect that users to use modulus(OO(X)) for that?

Yes, why not? From what I remember, this was intended as a scheme-sided analogue of `modulus(OO(X))`. But I don't remember the necessity to give this a shortcut name. So I think `modulus(OO(X))` is just as fine as it is not even longer to write.

Reply:
I would now vote for @simonbrandhorst's suggestion to completely omit this ambient_closure_ideal as a command, but document well in the scheme documentation how to get to such ideals (by the kernel(pullback(....)) variant for the inclusion in the ambient_affine_space and by the modulus(OO(X)) for a 'localized variant'.

Reply:
The only downside of this is that going through all these commands, one can not make use of previously cached stuff. Internally, this ideal sits there as a `saturated_ideal`. And in principle, one would be computing this saturation twice. But the knowledgeable user will also know about this way to get their hands on it and if in doubt, one has to program a dispatch for every type of ring, anyway. 

Long story short: If you don't advocate for keeping `ambient_closure_ideal`, we can remove it. Unless this breaks our code, of course.

Reply:
Fine with me, as long as the users' need for getting their hands on these ideals is satisfied by suitable documentation.

Reply:
I keep it for now, but I do not export it.

Reply:
> I keep it for now, but I do not export it.

I think that's reasonable for the time being. We should keep an eye on it as we progress. Because, from my experience, you get a clearer picture over time on how exactly you want and need to use things. If it proves to be really necessary in the long run, we will then eventually know better why and for what.